### ISSUE TITLE
Feature request: Support for <const> warnings in lua54 and lua54c

### ISSUE THREAD

**Issue by asherber** [lunarmodules/luacheck#116]
In Lua 5.4, luacheck should warn if a `<const>` variable is later assigned a new value.

```lua
local foo <const> = 5
foo = 6  -- should generate warning
```
